### PR TITLE
fix norm api doc, test=develop

### PR DESCRIPTION
**Pull request by frankwhzhang**
### PR types

Others 
### PR changes

APIs
### Describe

1. delete alias of group_norm and InstanceNorm
2. fix docs of spectral_norm
3. modify api of adagrad
4. enhance error message

### CODE REVIEW

File: python/paddle/optimizer/adagrad.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+from ..fluid import framework
+from ..fluid.framework import Variable
+
+__all__ = ["Adagrad"]
+
+
+class Adagrad(Optimizer):
+    """
+    The Adaptive Gradient optimizer (Adagrad for short) can adaptively assign
+    different learning rates to individual parameters.
```
Comment (jzhang533) [inline python/paddle/optimizer/adagrad.py]: update this brief introduction.

File: python/paddle/optimizer/adagrad.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    Related paper: `Adaptive Subgradient Methods for Online Learning and
+    Stochastic Optimization <http://www.jmlr.org/papers/volume12/duchi11a/duchi11a.pdf>`_.
+
+    The original paper does not have the ``epsilon`` attribute. It is added here
+    in our implementation as also proposed `Per-parameter adaptive learning rate
+    methods <http://cs231n.github.io/neural-networks-3/#ada>`_
+    for numerical stability to avoid the division by zero error.
+
+    Args:
+        learning_rate (float|Variable): The learning rate used to update ``Parameter``.
```
Comment (jzhang533) [inline python/paddle/optimizer/adagrad.py]: Variable改为Tensor

File: python/paddle/optimizer/adagrad.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            It can be a float value or a ``Variable`` with a float type.
+        epsilon (float, optional): A small float value for numerical stability.
+            The default value is 1e-06.
+        parameters (list, optional):  Iterable of ``Tensor`` names to update to minimize ``loss``. \
```
Comment (jzhang533) [inline python/paddle/optimizer/adagrad.py]: 参考一下`Adam`的这部分的描述

File: python/paddle/optimizer/adagrad.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	weight_decay (float|WeightDecayRegularizer, optional): The strategy of regularization. \
+	    It canbe a float value as coeff of L2 regularization or \
+	    :ref:`api_fluid_regularizer_L1Decay`, :ref:`api_fluid_regularizer_L2Decay`.
+	    If a parameter has set regularizer using :ref:`api_fluid_ParamAttr` already, \
+	    the regularization setting here in optimizer will be ignored for this parameter. \
+	    Otherwise, the regularization setting here in optimizer will take effect. \
+	    Default None, meaning there is no regularization.
+        grad_clip (GradientClipBase, optional): Gradient cliping strategy, it's an instance of 
+            some derived class of ``GradientClipBase`` . There are three cliping strategies 
+            ( :ref:`api_fluid_clip_GradientClipByGlobalNorm` , :ref:`api_fluid_clip_GradientClipByNorm` ,
```
Comment (jzhang533) [inline python/paddle/optimizer/adagrad.py]: 这里都改为2.0下的API（可以先不加ref来建立链接）

File: python/paddle/optimizer/adagrad.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+            The default value is 0.0.
+
+    Examples:
+        .. code-block:: python
+
+            import paddle
+            import numpy as np
+
+            paddle.disable_static()
+            inp = np.random.uniform(-0.1, 0.1, [10, 10]).astype("float32")
```
Comment (jzhang533) [inline python/paddle/optimizer/adagrad.py]: 可以直接用`paddle.rand`

File: python/paddle/optimizer/adagrad.py (hunk 3 above)
Comment (frankwhzhang) [reply python/paddle/optimizer/adagrad.py]: done

File: python/paddle/optimizer/adagrad.py (hunk 2 above)
Comment (frankwhzhang) [reply python/paddle/optimizer/adagrad.py]: done

File: python/paddle/optimizer/adagrad.py (hunk 5 above)
Comment (frankwhzhang) [reply python/paddle/optimizer/adagrad.py]: done

File: python/paddle/optimizer/adagrad.py (hunk 1 above)
Comment (frankwhzhang) [reply python/paddle/optimizer/adagrad.py]: done

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by zhiqiu — APPROVED** [review]
LGTM for spectral_norm without core.ops since it is used in static graph.

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**